### PR TITLE
[ISSUE #1262] Kafka Support for manual acknowledgement

### LINKED ISSUE #1262: [Enhancement] Kafka Support for Manual Ack (updateOffset API)
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

 Kafka Support for Manual Ack (updateOffset API)

### Describe the solution you'd like

Implement the updateOffset API

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by Markliniubility**
Fixes #1262.

### Motivation

*Kafka Support for manual acknowledgement: the previous version of Kakfa Connector auto commits on consuming the messages*

### Modifications

*Disabled the auto-commit, and implemented the manual commit with updateOffset() API*
*Pick the largest offset from the list of cloudevent given, then commit to that specific offset*

### Documentation

- Does this pull request introduce a new feature?
- Yes
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented) 
- Not yet documented
- If a feature is not applicable for documentation, explain why? 
- Will be documented after completion of development

### CODE REVIEW

File: eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/consumer/KafkaConsumerRunner.java [hunk 1]
```diff
@@ -37,22 +39,28 @@ public class KafkaConsumerRunner implements Runnable {
...
 
+    public long getOffset(CloudEvent cloudEvent) {
+        return cloudEventToOffset.getOrDefault(cloudEvent, 0L);
+    }
+
     @Override
     public void run() {
-        try {
+        try { // TODO: change it to manual ack
             while (!closed.get()) {
```
Comment (githublaohu) [inline eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/consumer/KafkaConsumerRunner.java:62]: 1. while里面加入一个try catch，catch里面打印日志就了
2.  在 forEach里面也加入一个try cath，一个消息报错。整个list就不执行了。这样不行吧

File: eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/consumer/KafkaConsumerRunner.java [hunk 2]
```diff
@@ -37,22 +39,28 @@ public class KafkaConsumerRunner implements Runnable {
...
+    public synchronized void setListener(EventListener listener) {
         this.listener = listener;
     }
 
+    public long getOffset(CloudEvent cloudEvent) {
+        return cloudEventToOffset.getOrDefault(cloudEvent, 0L);
+    }
+
     @Override
     public void run() {
```
Comment (githublaohu) [inline eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/consumer/KafkaConsumerRunner.java:60]: 如果这个线程是一直执行，请保证异常情况下不退出线程

File: eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/consumer/KafkaConsumerRunner.java [hunk 3]
```diff
@@ -64,29 +72,33 @@ public void commit(EventMeshAction action) {
...
+                    cloudEventToOffset.put(cloudEvent, rec.offset());
                     if (listener != null) {
                         listener.consume(cloudEvent, eventMeshAsyncConsumeContext);
                     }
                 });
             }
         } catch (WakeupException e) {
             // Ignore exception if closing
-            if (!closed.get()) throw e;
+            if (!closed.get()) {
```
Comment (githublaohu) [inline eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/consumer/KafkaConsumerRunner.java:97]: 感觉没有意义，closed会在while识别。

File: eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/consumer/ConsumerImpl.java [hunk 4]
```diff
@@ -108,9 +109,9 @@ public void subscribe(String topic) {
 
     public void unsubscribe(String topic) {
         try {
-            // Get the current subscription
+            // Kafka will unsubscribe *all* topic if calling unsubscribe, so we
             this.kafkaConsumer.unsubscribe();
-            topicsSet.add(topic);
+            topicsSet.remove(topic);
```
Comment (githublaohu) [inline eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/consumer/ConsumerImpl.java:114]: HashSet不是线程安全的，请保证topics.remove(),与115是线程安全的

File: eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/consumer/KafkaConsumerRunner.java (hunk 3 above)
Comment (Markliniubility) [reply eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/consumer/KafkaConsumerRunner.java]: 完成

File: eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/consumer/ConsumerImpl.java (hunk 4 above)
Comment (Markliniubility) [reply eventmesh-connector-plugin/eventmesh-connector-kafka/src/main/java/org/apache/eventmesh/connector/kafka/consumer/ConsumerImpl.java:114]: 添加了`synchronized`保证线程安全

**Review by xwm1992 — APPROVED** [review]
LGTM

**Pull request merged**